frontend/public/components/masthead.tsx

```diff
@@ -1,30 +1,155 @@
+    const {showAboutModal} = this.state;
+    return <React.Fragment>
+      <ActionsMenu
+        actions={[
+          {label: 'Documentation', callback: this.openDocumentation},
+          {label: 'About', callback: this.openAboutModal}]}
+        buttonClassName="nav-item-iconic"
+        noButton
+        noCaret
+        title={<i className="fa fa-question-circle-o co-masthead__help-icon" />} />
+      <AboutModal className={classNames('co-masthead__modal', {'co-masthead__modal--upstream': backgroundImg})} logo={logoImg} altLogo={logoAlt} productTitle={title} show={showAboutModal} onHide={this.closeAboutModal}>
+        <strong>About</strong>
+        <p>{title === 'OKD' ? 'OKD' : 'OpenShift'} is Red Hat&apos;s container application platform that allows developers to quickly develop, host, and scale applications in a cloud environment.</p>
+        {(openshiftVersion || kubernetesVersion) &&
+          <React.Fragment>
+            <strong>Version</strong>
+            <AboutModal.Versions className="co-masthead__modal--version">
+              {openshiftVersion && <AboutModal.VersionItem label={`${title === 'OKD' ? 'OKD' : 'OpenShift'} Master`} versionText={openshiftVersion} />}
+              {kubernetesVersion && <AboutModal.VersionItem label="Kubernetes Master" versionText={kubernetesVersion} />}
+            </AboutModal.Versions>
+          </React.Fragment>}
+      </AboutModal>
+    </React.Fragment>;
+  }
+}
+
+const BrandingDetails = () => {
+  let backgroundImg, logoImg, logoAlt, modalLogoImg, modalLogoAlt, productTitle;
+
+  // Webpack won't bundle these images if we don't directly reference them, hence the switch
+  switch ((window as any).SERVER_FLAGS.branding) {
+    case 'ocp':
+      backgroundImg = true;
+      logoImg = ocpLogoImg;
+      logoAlt = 'OpenShift Container Platform';
+      modalLogoImg = rhLogoImg;
+      modalLogoAlt = 'Red Hat';
+      productTitle = <React.Fragment>Red Hat<sup>&reg;</sup> OpenShift Container Platform</React.Fragment>;
+      break;
+    case 'online':
+      backgroundImg = true;
+      logoImg = onlineLogoImg;
+      logoAlt = 'OpenShift Online';
+      modalLogoImg = rhLogoImg;
+      modalLogoAlt = 'Red Hat';
+      productTitle = <React.Fragment>Red Hat<sup>&reg;</sup> OpenShift Online</React.Fragment>;
+      break;
+    case 'dedicated':
+      backgroundImg = true;
+      logoImg = dedicatedLogoImg;
+      logoAlt = 'OpenShift Dedicated';
+      modalLogoImg = rhLogoImg;
+      modalLogoAlt = 'Red Hat';
+      productTitle = <React.Fragment>Red Hat<sup>&reg;</sup> OpenShift Dedicated</React.Fragment>;
+      break;
+    default:
+      backgroundImg = false;
+      logoImg = okdLogoImg;
+      logoAlt = 'OKD';
+      modalLogoImg = okdModalImg;
+      modalLogoAlt = 'OKD';
+      productTitle = 'OKD';
+  }
+
+  return ({'backgroundImg': backgroundImg, 'logoImg': logoImg, 'logoAlt': logoAlt, 'modalLogo': modalLogoImg, 'modalLogoAlt': modalLogoAlt, 'productTitle': productTitle});
```

Comment:
Use [object literal shorthand notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015) instead of specifying both keys and values.